src/appTypeDefs.ts

```diff
@@ -0,0 +1,49 @@
+export interface PostComment {
+  postId: number;
+  id: number;
+  name: string;
+  email: string;
+  body: string;
+}
+
+export interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+}
+
+export interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+  address: Address;
+  phone: string;
+  website: string;
+  company: Company;
+}
+
+type Address = {
+  street: string;
+  suite: string;
+  city: string;
+  zipcode: string;
+  geo: Geo;
+};
+
+type Geo = {
+  lat: string;
+  lng: string;
+};
+
+type Company = {
+  name: string;
+  catchPhrase: string;
+  bs: string;
+};
+
+export interface PreparedPost extends Post {
+  postComments: PostComment[];
+  user?: User;
```

Comment:
`user` is not optional. It should be there but if it's value is falsy it should be `null`. Just like you assigned `null` if user is not found by `user.id` in your helper function.

Reply:
Fixed it thank you.